src/lib/factories.js

```diff
@@ -2,86 +2,146 @@ import _ from 'lodash'
 import cx from 'classnames'
 import React, { cloneElement, isValidElement } from 'react'
 
+// ============================================================
+// Factory Utilities
+// ============================================================
 /**
- * Merges props and classNames.
+ * Generate a short hash code from a string.  Simplified and faster version of:
+ * http://stackoverflow.com/questions/7616461/generate-a-hash-from-string-in-javascript-jquery
  *
- * @param {object} defaultProps A props object
- * @param {object} props A props object
- * @returns {object} A new props object
+ * @param {string} str Any string
+ * @returns {number}
  */
-const mergePropsAndClassName = (defaultProps, props) => {
-  const { childKey, ...newProps } = { ...defaultProps, ...props }
-
-  if (_.has(props, 'className') || _.has(defaultProps.className)) {
-    newProps.className = cx(defaultProps.className, props.className) // eslint-disable-line react/prop-types
+const getHashCode = function hashCode(str) {
+  const { length } = str
+  if (!str || length === 0) return 0
+  let hash = 0
+  for (let i = 0; i < length; i++) {
+    hash = hash * 31 + str.charCodeAt(i) | 0
   }
+  return hash
+}
```

Comment:
When we generate keys we make a prop string, then we make a hash of that.  The only added benefit of the hash is having very short keys.  The downside is more processing at render time.  We could just leave the shallow stringified props as the key.  DOM weight is not an issue here as the key is in memory only.

I'm on the brink of dropping it already, LMK your thoughts.